### PR TITLE
bpo-29406: Prevent SSL socket leaking in asyncio

### PR DESCRIPTION
**Pull request by asvetlov**
https://bugs.python.org/issue29406

Supersedes #2269

### CODE REVIEW

File: Lib/asyncio/sslproto.py [hunk 1]
```diff
@@ -550,6 +553,15 @@ def _start_shutdown(self):
...
             self._write_appdata(b'')
 
+        if self._shutdown_timeout is not None:
+            self._shutdown_timeout_handle = self._loop.call_later(
+                self._shutdown_timeout, self._on_shutdown_timeout)
+
+    def _on_shutdown_timeout(self):
+        if self._transport is not None:
+            self._fatal_error(
+                futures.TimeoutError(), 'Can not complete shitdown operation')
```
Comment (arthurdarcet) [inline Lib/asyncio/sslproto.py]: typo

Comment (asvetlov) [reply Lib/asyncio/sslproto.py]: Good catch, thanks!

**Comment by 1st1** [comment]
Should this PR be closed?

**Comment by asvetlov** [comment]
Please let me review first.

**Comment by 1st1** [comment]
I thought you've already merged something that fixes this, no?

**Comment by 1st1** [comment]
What's the status of this?

**Comment by dimaqq** [comment]
ping...

**Comment by vstinner** [comment]
I removed the " needs backport to 3.6" label, the 3.6 branch no long accept bugfixes (only security fixes are accepted): https://devguide.python.org/#status-of-python-branches

**Comment by mcauto** [comment]
Why closed?

**Pull request closed without merging**